testutil/containschecker.go

```diff
@@ -150,3 +150,69 @@ func (c *deepContainsChecker) Check(params []interface{}, names []string) (resul
 		return false, fmt.Sprintf("%T is not a supported container", container)
 	}
 }
+
+type deepContainerMatchChecker struct {
+	*check.CheckerInfo
+}
+
+// DeepContainerMatches checks if two containers contain the same elements in
+// the same number (but possibly different order) using DeepEqual. The container
+// can be an array, a slice or a map.
+var DeepContainerMatches check.Checker = &deepContainerMatchChecker{
+	&check.CheckerInfo{Name: "DeepContainerMatchChecker", Params: []string{"container1", "container2"}},
+}
+
+func (c *deepContainerMatchChecker) Check(params []interface{}, _ []string) (bool, string) {
+	container1 := reflect.ValueOf(params[0])
+	container2 := reflect.ValueOf(params[1])
+
+	if container1 == container2 {
+		return true, ""
+	}
+
+	if container1.Kind() != container2.Kind() {
+		return false, fmt.Sprintf("containers are of different type: %s != %s", container1.Kind(), container2.Kind())
+	}
+
+	if container1.Len() != container2.Len() {
+		return false, fmt.Sprintf("containers have different lengths: %d != %d", container1.Len(), container2.Len())
+	}
```

Comment:
Is there a way to also print the contents of the two containers? That would be helpful in debugging.

Reply:
I can change it back to printing the differences only (length, type, index x is unmatched, etc) and